sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasBuilder.cs

```diff
@@ -344,16 +344,22 @@ private static string GetCanonicalName(string account, string fileSystemName, st
         /// </summary>
         private void EnsureState()
         {
-            if (ExpiresOn == default)
+            if (string.IsNullOrEmpty(Identifier))
```

Comment:
See https://github.com/Azure/azure-sdk-for-net/issues/8840

This was a bug, it also needs addressed in Blob, Files.Shares, and Queues.